pkg/sink/sink.go

```diff
@@ -145,10 +152,16 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 		}
 	}
 
-	// TODO: Do we really need to return the entire body back???
 	response.WriteHeader(code)
-	fmt.Fprintf(response, "EventListener: %s in Namespace: %s handling event (EventID: %s) with payload: %s and header: %v",
-		r.EventListenerName, r.EventListenerNamespace, string(eventID), string(event), request.Header)
+	response.Header().Set("Content-Type", "application/json")
+	body := Response{
+		EventListener: r.EventListenerName,
+		Namespace:     r.EventListenerNamespace,
+		EventID:       eventID,
+	}
+	if err := json.NewEncoder(response).Encode(body); err != nil {
+		eventLog.Errorf("failed to write back sink response: %w", err)
```

Comment:
do we need to do anything else in this case, e.g. set the header or the result with an error (looks like above error handling does this?)

Reply:
Basically there are two cases where we'd get an error here:
1. The JSON encoding fails (this should basically never happen since we are setting just setting static values to a struct that we know of)\

2. Something goes wrong while sending the actual body out e.g. the underlying connection is closed

In both cases, the issue is with sending the response body back and not with the actual processing of the event. And in 2. we cannot guarantee that we can even successfully write the status code successfully.

The GitHub webhook best practices recommends sending a 500 only if it is a catastrophic error which I don't think this is since the actually trigger processing goes through and we only fail when writing out the response body.

Reply:
ah okay makes sense! thanks for explaining :D

do you think this could be worth a comment in the code explaining it? (no worries if not!)